internal/engine/compiler/impl_amd64.go

```diff
@@ -4970,19 +4972,9 @@ func (c *amd64Compiler) compileReservedStackBasePointerInitialization() {
 		amd64ReservedRegisterForCallEngine, callEngineGlobalContextValueStackElement0AddressOffset,
 		amd64ReservedRegisterForStackBasePointerAddress)
 
-	// Since initializeReservedRegisters is called at the beginning of function
-	// calls (or right after they return), we have free registers at this point.
-	tmpReg, _ := c.locationStack.takeFreeRegister(registerTypeGeneralPurpose)
-
 	// next we move the base pointer (callEngine.stackBasePointer) to the tmp register.
-	c.assembler.CompileMemoryToRegister(amd64.MOVQ,
-		amd64ReservedRegisterForCallEngine, callEngineValueStackContextStackBasePointerOffset,
-		tmpReg,
-	)
-
-	c.assembler.CompileMemoryWithIndexToRegister(
-		amd64.LEAQ,
-		amd64ReservedRegisterForStackBasePointerAddress, 0, tmpReg, 8,
```

Comment:
these two instructions are now consolidated to one ADDQ instruction, plus we don't need tmpReg anymore!